src/fmeval/eval_algorithms/factual_knowledge.py

```diff
@@ -52,167 +103,90 @@ def __post_init__(self):
+
     def __init__(self, eval_algorithm_config: FactualKnowledgeConfig = FactualKnowledgeConfig()):
-        """Default constructor
+        """FactualKnowledge initializer.
 
-        :param eval_algorithm_config: Factual knowledge eval algorithm config.
+        :param eval_algorithm_config: Factual knowledge evaluation algorithm config.
         """
         super().__init__(eval_algorithm_config)
-        self.eval_name = FACTUAL_KNOWLEDGE
-        self._eval_algorithm_config = eval_algorithm_config
+        self.pipeline = TransformPipeline(
+            [FactualKnowledgeScore(target_output_delimiter=eval_algorithm_config.target_output_delimiter)]
+        )
 
     def evaluate_sample(self, target_output: str, model_output: str) -> List[EvalScore]:  # type: ignore[override]
-        """
-        Factual knowledge evaluate sample.
-
-        Given an input prompt e.g., "London is the capital of" and expected answers(target_output) like
-        ["United Kingdom", "England"], if the model is able to arrive at the correct completion(model_output).
-        Generating any of the expected answers is considered a correct completion.
-        Since models might generate long outputs, this evaluation does not look for an exact match.
-        It considers the completion to be correct if the answer is contained within the model output generated.
+        """Compute the factual knowledge score on a single sample.
 
-        :param target_output: The expected responses from the model
-        :param model_output: The output of a model that we want to evaluate.
-        :return: list of EvalScore object
+        :param target_output: The expected responses from the model.
+        :param model_output: The output of the model being evaluated.
+        :return: A single-element list containing an EvalScore corresponding to the factual knowledge score.
         """
-        if target_output is None:
-            raise EvalAlgorithmClientError(
-                "Missing required input: target_output, for FactualKnowledge evaluate_sample"
-            )
-        if model_output is None:
-            raise EvalAlgorithmClientError("Missing required input: model_output, for FactualKnowledge evaluate_sample")
-
-        return [
-            EvalScore(
-                name=self.eval_name,
-                value=self._get_score(target_output=target_output, model_output=model_output),
-            )
-        ]
+        sample = {
+            DatasetColumns.TARGET_OUTPUT.value.name: target_output,
+            DatasetColumns.MODEL_OUTPUT.value.name: model_output,
+        }
+        result = self.pipeline.execute_record(sample)
+        return [EvalScore(name=FACTUAL_KNOWLEDGE, value=result[FACTUAL_KNOWLEDGE])]
 
     def evaluate(
         self,
         model: Optional[ModelRunner] = None,
         dataset_config: Optional[DataConfig] = None,
         prompt_template: Optional[str] = None,
+        num_records: int = 100,
```

Comment:
I recall it was intentional to have default `num_records` be 300 for factual knowledge, could you double check with science team on this change?

Reply:
Thanks for pointing this out! Will do.